src/main/java/co/inventorsoft/springresttask/service/impl/EmailDeliveryServiceImpl.java

```diff
@@ -0,0 +1,64 @@
+package co.inventorsoft.springresttask.service.impl;
+
+import co.inventorsoft.springresttask.controller.dto.EmailDto;
+import co.inventorsoft.springresttask.service.EmailDeliveryService;
+import co.inventorsoft.springresttask.service.mapper.EmailMapper;
+import co.inventorsoft.springresttask.service.model.Email;
+import co.inventorsoft.springresttask.service.repository.EmailRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class EmailDeliveryServiceImpl implements EmailDeliveryService {
+
+    private final EmailMapper emailMapper;
+    private final EmailRepository emailRepository;
+
+    @Override
+    public EmailDto getEmail(int id) {
+        return emailMapper.mapModelToDto(emailRepository.getEmail(id));
+    }
+
+    @Override
+    public List<EmailDto> listEmails() {
+        return emailRepository.listEmails()
+                .stream()
+                .map(emailMapper::mapModelToDto)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public EmailDto createEmail(EmailDto emailDto) {
+        log.info("Email with id {} was created", emailDto.getId());
+
+        java.util.Date dt = new java.util.Date();
+        java.text.SimpleDateFormat sdf =
```

Comment:
Have a look at Java time API. There you will have to pass dates as ISO strings. In REST API consistency is crucial and the server should not be responsible for data formatting, it's client's job